File: src/main/java/com/djdapz/stubhub/config/UrlConfigImpl.java
package com.djdapz.stubhub.config;

import lombok.SneakyThrows;
import lombok.experimental.FieldDefaults;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

import java.net.URL;

import static lombok.AccessLevel.PRIVATE;

@FieldDefaults(level = PRIVATE, makeFinal = true)
@Configuration
public class UrlConfigImpl implements UrlConfig {

    private final String baseUrl;

    private final String listingsPath;

    public UrlConfigImpl(
            @Value("${stubhub.url}") String baseUrl,
            @Value("${stubhub.path.listing}") String listingsPath
    ) {
        this.baseUrl = baseUrl;
        this.listingsPath = listingsPath;
    }

    @Override
    @SneakyThrows
    public URL getStubhubListingUrl() {
        return new URL(baseUrl + listingsPath);
    }
}

File: src/main/java/com/djdapz/stubhub/config/UrlConfigImpl.kt
package com.djdapz.stubhub.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import java.net.URL

@Configuration
class UrlConfigImpl(
        @Value("\${stubhub.url}") private val baseUrl: String,
        @Value("\${stubhub.listing.path}") private val listingsPath: String,
        @Value("\${stubhub.listing.page-size}") private val listingsPageSize: Int

) : UrlConfig {
    override val pageSize: Int
        get() = listingsPageSize

    override val stubhubListingUrl: URL
        get() = URL(baseUrl + listingsPath)
}
